DataFormats/PatCandidates/interface/PackedCandidate.h

```diff
@@ -443,20 +397,37 @@ namespace pat {
       if (numberOfPixelHits_ > trackPixelHitsMask) numberOfPixelHits_ = trackPixelHitsMask;
       int numberOfStripHits_ = tk.hitPattern().numberOfValidHits() - numberOfPixelHits_ - numberOfPixelLayers_ - numberOfStripLayers_;
       if (numberOfStripHits_ > trackStripHitsMask) numberOfStripHits_ = trackStripHitsMask;
+
       packedHits_ = (numberOfPixelHits_&trackPixelHitsMask) | (numberOfStripHits_ << trackStripHitsShift);
- 
+    }
+  
+    virtual void setTrackProperties( const reco::Track & tk, const reco::Track::CovarianceMatrix & covariance,int quality,int covarianceVersion) {
+      covarianceVersion_ = covarianceVersion ;
+      covarianceSchema_ = quality ;
+      normalizedChi2_ = tk.normalizedChi2();
+      //
+      setHits(tk); //tk.hitPattern().numberOfValidPixelHits(),tk.hitPattern().numberOfValidHits());
       packBoth();
+      packCovariance(covariance,false); //TODO: check if these 2 lines make sense as the m_ is protected behind the Track atomic pointer...
+      //unpackTrk();         // 
     }
 
-    virtual void setTrackProperties( const reco::Track & tk ) {
-	setTrackProperties(tk,tk.covariance());
+    virtual void setTrackHits( const reco::Track & tk ) {
+	setHits(tk);
+    }	
+    virtual void setTrackProperties( const reco::Track & tk,int quality,int covarianceVersion ) {
```

Comment:
here and other places it would be nice to keep doxygen-style comments to keep inline interface documentation